third_party/move/move-bytecode-verifier/src/signature_v2.rs

```diff
@@ -173,6 +173,18 @@ fn check_ty<const N: usize>(
                 param_constraints,
             )?;
         },
+        Function(args, result, abilities) => {
+            assert_abilities(*abilities, required_abilities)?;
+            for ty in args.iter().chain(result.iter()) {
+                check_ty(
```

Comment:
Do we actually need to check the arugments abilities here? The arguments ability should be irrelavant to the closure's ability?

Reply:
Good finding! Yes, you are right, this check is unnecessary. Removed.